### PR TITLE
fix(backing): add user friendly error message for backing image upload

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: [longhorn/longhorn#3294](https://github.com/longhorn/longhorn/issues/3294)

### CODE REVIEW

File: pkg/sync/service.go [hunk 1]
```diff
@@ -380,6 +381,7 @@ func (s *Service) doDownloadFromURL(request *http.Request) (err error) {
 func (s *Service) UploadFromRequest(writer http.ResponseWriter, request *http.Request) {
 	err := s.doUploadFromRequest(request)
 	if err != nil {
+		err = errors.Wrapf(err, "please make sure the file is qcow2 or raw image")
```
Comment (shuo-wu) [inline pkg/sync/service.go:384]: Is this message necessary? It misleads users that the failure is caused by incorrect file format but actually there are other kinds of errors.

Comment (ChanYiLin) [reply pkg/sync/service.go:384]: Yes you are right, put it back to the original error

**Comment by shuo-wu** [comment]
Just checked the modification, to be honest, I don't think it's necessary. At least the statement you added is not related to the error...

**Comment by ChanYiLin** [comment]
@shuo-wu 
actually I agree with you
We actually don't check the file type when uploading backing image
we only care if it is a multiple of 512 bytes 

I'll discuss with @innobead for the original issue

**Comment by innobead** [comment]
@ChanYiLin remember to assign yourself to the assignee to prevent the issue from becoming stale after days. 

cc @longhorn/dev

**Comment by ChanYiLin** [comment]
After discussed with @innobead, will improve the user experience from other perspective instead of this error message

**Pull request closed without merging**